### DISCUSSION TITLE
non style-generating component tokens

### DISCUSSION THREAD

**Discussion by engegu** [six7/figma-tokens#962] [Ideas]
**TLDR;** 
I want to be able to use component specific tokens in Figma, but not have them generating colour styles. Objects that has a component token applied should display the semantic token it refrences.

Our design system aims to be easy to use and readable for newly onboarded designers. We have opted to use only our semantic tokens, i.e `background.action.default` in Figma. Component tokens referencing these i.e; `button.primary.background.default` are then up to the developer to make on their end if they think they will need them.

**What I want:**
* An easy to read and navigate colour system in Figma, the semantic level is perfect. I do not want hundreds of component specific colour styles.
* To be able to use component specific tokens in Figma, if I so choose.

**My idea**
A kind of similar concept to "treat as source", I can mark a set of components as "component tokens" or something, and those components will style the objects they are applied to with the style of the semantic colour it refrences. or just a hex if there is none.

Had an interesting chat with Mike earlier and agreed to post this here. What do you guys think?

**Comment by batsper** [comment]
+1!

We are creating the same type of system with token tiers. Our semantic tokens create Figma styles and said styles are used by designers. Developers use component specific tokens. In order for our components to be truly 1:1 we also build the components in Figma using the component tokens and connect the styles that our semantic tokes created afterwards. A bit cumbersome and a bit of a workaround, so this would be absolutely wonderful!

**Comment by StefanIvanov** [comment]
+1 we are looking to achieve the exact same result

**Comment by six7** [comment]
I do like the idea, however we'd need to restrict what happens when non-pure references are used.

I could imagine this to tie in nicely with the Themes feature where we will be able to connect a token to a style. Essentially it would be a setting/mode where a certain token set would be set to not create styles itself, but defer the style to its source. 

It would mean however that as soon as you create a style in any other way than a pure reference, it would fallback to hex values. (Eg alpha modifications, gradients or hex values would all apply the hex, while a pure {reference} would instead apply the reference-style.

For a token `components.card.background` that has a value of `{theme.bg.subtle}` When we attach styles to a theme in the file where `theme.bg.subtle` exists and a token set is set to this mode, the plugin attaches the style `theme.bg.subtle` to the token `components.card.background`.

Now whenever you apply the  `components.card.background` token the plugin will instead apply the `theme.bg.subtle` style.

It might also work without having to attach styles, as long as `theme.bg.subtle` has an attached style.

**Reply by engegu** [reply]
Yeah, thats exactly how I think it should work.
